App/Views/RenderView.swift

```diff
@@ -325,17 +378,48 @@ extension RenderView {
             }
         }
     }
-    
-    func loadLottiePlayer() {
+
+    // MARK: - Private Helpers
+
+    /// Helper to evaluate Awful JavaScript functions with consistent error handling.
+    private func evalAwfulFunction(_ functionName: String) {
         Task {
             do {
-                try await webView.eval("if (window.Awful) Awful.loadLotties()")
+                try await webView.eval("if (window.Awful) { Awful.\(functionName)(); }")
             } catch {
-                self.mentionError(error, explanation: "could not evaluate loadLotties")
+                self.mentionError(error, explanation: "could not evaluate \(functionName)")
             }
         }
     }
 
+    // MARK: - Image Loading Methods
+
+    /// Applies timeout detection to images that are loading immediately (first 10 images).
+    ///
+    /// Monitors download progress and replaces stalled images with dead image badges.
+    /// This function should be called after the page has loaded to set up monitoring
+    /// for the initial batch of images.
+    func applyTimeoutToLoadingImages() {
+        evalAwfulFunction("applyTimeoutToLoadingImages")
+    }
+
+    /// Sets up lazy loading for deferred images (images 11+).
+    ///
+    /// Uses IntersectionObserver to load images as they approach the viewport.
+    /// Images are loaded approximately 600px before they would be visible to the user
+    /// for a seamless scrolling experience.
+    func setupImageLazyLoading() {
```

Comment:
This might be dead code; I haven't pulled down the branch but there's no `setupImageLazyLoading` elsewhere in this PR that I can see.